Notes.Infrastructure/Notes.Infrastructure/Repositories/NoteRepository.cs

```
@@ -24,5 +24,16 @@ public async Task UpdateAsync(Note note)
             var filter = Builders<Note>.Filter.Eq(n => n.Id, note.Id);
             await _notes.ReplaceOneAsync(filter, note);
         }
+
+        public async Task<List<Note>> SearchAsync(string searchTerm)
+        {
+            if (searchTerm == string.Empty)
```

This should probably throw an exception than return everything.